src/SSHDebugPS/PipeConnection.cs

```diff
@@ -108,35 +108,50 @@ public override bool IsLinux()
             return commandOutput.StartsWith("Linux", StringComparison.OrdinalIgnoreCase);
         }
 
-        public bool TryGetUsername(out string username)
+        public bool TryGetSystemInformation(out SystemInformation systemInformation)
         {
-            username = string.Empty;
+            systemInformation = null;
+
+            string username = string.Empty;
             string command = "id -u -n";
             string commandOutput;
             string errorMessage;
             if (ExecuteCommand(command, Timeout.Infinite, throwOnFailure: false, commandOutput: out commandOutput, errorMessage: out errorMessage))
             {
 
                 username = commandOutput;
-                return true;
             }
 
-            return false;
+            string architecture = string.Empty;
+            command = "uname -m";
+            if (ExecuteCommand(command, Timeout.Infinite, throwOnFailure: false, commandOutput: out commandOutput, errorMessage: out errorMessage))
+            {
+
+                architecture = commandOutput;
+            }
+
+            if (string.IsNullOrWhiteSpace(username) || string.IsNullOrWhiteSpace(architecture))
+            {
+                return false;
+            }
+
+            systemInformation = new SystemInformation(username, architecture);
+            return true;
         }
 
         public override List<Process> ListProcesses()
         {
-            string username;
-            TryGetUsername(out username);
+            SystemInformation systemInformation;
+            TryGetSystemInformation(out systemInformation);
```

Comment:
>TryGetSystemInformation [](http://example.com/codeflow?start=12&length=23)

Suggestion: `if (!...) { systemInformation = new SystemInformation(string.Empty, string.Empty); }`

Reply:
Changed it to always return a system information, however it can return `new SystemInformation(string.Empty, string.Empty);`